e2e/testdata/test-complex-service/main.go

```diff
@@ -59,68 +44,89 @@ func New() (*Client, error) {
 		return nil, fmt.Errorf("client: connection error: %s", err)
 	}
 
-	return &Client{
-		ExecutionClient: pb.NewExecutionClient(conn),
-		EventClient:     pb.NewEventClient(conn),
-		InstanceHash:    instanceHash,
-	}, nil
+	return runner.NewRunnerClient(conn), nil
 }
 
-// SendEvent creates a new event.
-func (c *Client) SendEvent(key string) {
+func register(client runner.RunnerClient) (string, error) {
+	msg := os.Getenv(envMesgMsg)
+	if msg == "" {
+		return "", fmt.Errorf("env %q is empty", envMesgMsg)
+	}
+	signature := os.Getenv(envMesgSignature)
+	if signature == "" {
+		return "", fmt.Errorf("env %q is empty", envMesgSignature)
+	}
+
+	resp, err := client.Register(context.Background(), &runner.RegisterRequest{
+		Msg:       msg,
+		Signature: signature,
+	})
+	if err != nil {
+		return "", err
+	}
+	return resp.Token, nil
+}
+
+// sendEvent creates a new event.
+func sendEvent(client runner.RunnerClient, token string, key string) {
 	log.Println("sending event:", key)
-	if _, err := c.EventClient.Create(context.Background(), &pb.CreateEventRequest{
-		InstanceHash: c.InstanceHash,
-		Key:          key,
-	}); err != nil {
+	if _, err := client.Event(context.Background(), &runner.EventRequest{
+		Key: key,
+	}, grpc.PerRPCCredentials(runner.NewTokenCredential(token))); err != nil {
```

Comment:
this is the way to send the credential to a specific request.
There is a way to send the credential to every request of the client, but as Register doesn't need it and the token is only obtained after calling register, it will be a weird design.